src/main/java/Fenix.java

```diff
@@ -0,0 +1,197 @@
+        switch (command) {
+        case "bye":
+            bidFarewell();
+            return;
+        case "list":
+            showAllTasks(false);
+            break;
+        case "mark":
+            markAsDone(words);
+            break;
+        case "unmark":
+            unmarkAsDone(words);
+            break;
+        case "todo":
+        case "deadline":
+        case "event":
+            processTasks(userInput);
+            break;
+        default:
+            System.out.println("Please provide a valid command");
+        }
+        acceptUserInput();
+    }
+
+    public void bidFarewell() {
+        System.out.println(FAREWELL);
+        System.out.println(HORIZONTAL_LINE_USER_COMMAND);
+        scanner.close();
+    }
+
+    public void showAllTasks(boolean isModified) {
+        String index;
+        String task;
+        String extraSpace = (isModified ? "\t" : "");
+        String space = extraSpace + "\t";
+        for (int i = 0; i < taskArray.length && !isNullElement(i); i += 1) {
+            index = (i + 1) + ". ";
+            task = taskArray[i].toString();
+            System.out.println(space + index + task);
+        }
+    }
+
+    public boolean isNullElement(int i) {
+        return taskArray[i] == null;
+    }
+
+    private void markAsDone(String[] words) {
+        try {
+            String taskNumber = words[1];
+            if (isValidTaskNumber(taskNumber)) {
+                int taskIndex = Integer.parseInt(taskNumber) - 1;
+                markTaskAsDone(taskIndex);
+            } else {
+                System.out.println("Please provide a valid task number to mark");
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Please provide a task");
+        }
+    }
+
+    private boolean isValidTaskNumber(String input) {
+        try {
+            int index = Integer.parseInt(input);
+            return index > 0 && index <= taskNumber;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    private void unmarkAsDone(String[] words) {
+        try {
+            String taskNumber = words[1];
+            if (isValidTaskNumber(taskNumber)) {
+                int taskIndex = Integer.parseInt(taskNumber) - 1;
+                unmarkTaskAsDone(taskIndex);
+            } else {
+                System.out.println("Please provide a valid task number to unmark");
+            }
+        } catch (ArrayIndexOutOfBoundsException aiobe) {
+            System.out.println("Please provide a task");
+        }
+    }
+
+    public void markTaskAsDone(int taskNumber) {
+        System.out.println("Task successfully completed. A job well executed.");
+        taskArray[taskNumber].markAsDone();
+        System.out.println(HORIZONTAL_LINE_FENIX_MODIFICATION);
+        showAllTasks(true);
+        System.out.println(HORIZONTAL_LINE_FENIX_MODIFICATION);
+    }
+
+    public void unmarkTaskAsDone(int taskNumber) {
```

Comment:
Similarly unmarkTaskAsDone sounds a bit confusing, perhaps you could use markTaskAsUndone instead.